gcc/common/config/riscv/riscv-common.c

```diff
@@ -642,8 +644,9 @@ riscv_subset_list::handle_implied_ext (const char *ext,
 	continue;
 
       /* TODO: Implied extension might use different version.  */
-      add (implied_info->implied_ext, major_version, minor_version,
-	   explicit_version_p);
+      if (xlen () >= implied_info->xlen)
```

Comment:
If this logic will not be reverted, it should check either equality or if this xlen is available (in case of rv32-only Zeds).
```c
if (!implied_info->xlen // any xlen
    || (xlen() & implied_info->xlen))
```

Reply:
By "available" I meant that `implied_info->xlen[_requirement]` is a mask of legal xlens (e.g. `16 | 32` passes this check on xlen=16 and xlen=32 but not xlen=64) -- this is why I suggested a bit-wise check.
Anyway, everything will blow up with any xlen not in [32, 64] so it should be fine.

Reply:
I agree. but since the xlen checking for instructions in [riscv-binutils](https://github.com/riscv/riscv-binutils-gdb/blob/riscv-binutils-2.37/opcodes/riscv-dis.c#L507) works in this way, so I think maybe we can just leave it there.